core/src/main/java/com/datastrato/gravitino/catalog/CatalogManager.java

```diff
@@ -135,6 +137,19 @@ public <R> R doWithTopicOps(ThrowableFunction<TopicCatalog, R> fn) throws Except
           });
     }
 
+    public <R> R doWithPartitionOps(
+        NameIdentifier tableIdent, ThrowableFunction<SupportsPartitions, R> fn) throws Exception {
+      return classLoader.withClassLoader(
+          cl -> {
+            Preconditions.checkArgument(
+                asTables() != null, "Catalog does not support table operations");
+            Table table = asTables().loadTable(tableIdent);
+            Preconditions.checkArgument(
+                table.supportPartitions() != null, "Table does not support partition operations");
```

Comment:
UnsupportedOperationException

Reply:
So should I remove this check?

Reply:
Perhaps it can avoid the implementer returning null.

Reply:
if it will never return null, you don't need to handle this, otherwise, you need to think about how to handle UnsupportedOperationException

Reply:
> if it will never return null, you don't need to handle this,

Whether null will be returned depends on the developer of the catalog. The current implementation is more like defensive programming.

> you need to think about how to handle UnsupportedOperationException

Since `UnsupportedOperationException` is a `RuntimeException`, I think we could just throws it out